src/helpers/types.ts

```diff
@@ -83,7 +83,7 @@ export function convertToType(Target: any, data?: object): object | undefined {
     return;
   }
   // Prevent unecessary conversion
-  if (data instanceof Target) {
+  if (typeof Target === "object" || data instanceof Target) {
```

Comment:
Maybe it's better to have just `if (data instanceof Target)` but placed in line 97?
Then all the scalar and simple types cases will be handled.